webpack.common.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+const MiniCssExtractPlugin = require('mini-css-extract-plugin');
+const path = require('path');
+
+module.exports = {
+  entry: './jujugui/static/gui/src/app/init.js',
+  output: {
+    path: path.resolve(__dirname, 'jujugui/static/gui/build/app'),
+    filename: 'init-pkg.js',
+    library: 'JujuGUI',
+    libraryTarget: 'var'
+  },
+  module: {
+    rules: [
```

Comment:
Can you please add code comments to each rule explaining what a rule does and why it's there. Webpack configs in projects get unwieldy fast and are quite difficult to read. Also in the prod, dev, test versions.